modules/collect/src/main/java/com/opengamma/strata/collect/io/ArrayByteSource.java

```diff
@@ -221,6 +255,24 @@ public MetaBean metaBean() {
     return Meta.META;
   }
 
+  @Override
+  public Optional<String> getFileName() {
+    return Optional.ofNullable(fileName);
+  }
+
+  /**
+   * Returns an instance with the file name updated.
+   * <p>
+   * If a path is passed in, only the file name is retained.
+   * 
+   * @param fileName the file name, an empty string can be used to remove the file name
+   * @return a source with the specified file name
+   */
+  public ArrayByteSource withFileName(String fileName) {
+    ArgChecker.notNull(fileName, "fileName");
+    return new ArrayByteSource(array, Paths.get(fileName).getFileName().toString());
```

Comment:
Uhhh I think this uses the local default filesystem to resolve that filename, which isn't necessarily what you want.